tox.ini

```diff
@@ -77,9 +77,17 @@ commands =
 
 [testenv:lint]
 description = Performs linting, style checks
-skip_install = true
-sitepackages = false
+allowlist_externals =
+    bash
+extras =
+    azure
+    docker
+    ec2
+    gce
+    podman
+    vagrant
 deps =
     pre-commit
 commands =
+    bash {toxinidir}/tools/generate-templates.sh
     pre-commit run -a
```

Comment:
```suggestion
    pre-commit run -a
    rm -rf test/roles
```

Reply:
You are right. The .gitignore doesn't work as expect.

Let us clean up the temporary files explicitly at the end of test.